src/Umbraco.Core/Services/MediaEditingService.cs

```diff
@@ -51,42 +48,59 @@ internal sealed class MediaEditingService
     {
-        Attempt<ContentEditingOperationStatus> result = await MapUpdate(content, updateModel);
+        Attempt<ContentEditingOperationStatus> result = await MapUpdate(media, updateModel);
         if (result.Success == false)
         {
-            return Attempt.FailWithStatus(result.Result, content);
+            return Attempt.FailWithStatus(result.Result, media);
         }
 
         var currentUserId = await GetUserIdAsync(userKey);
-        ContentEditingOperationStatus operationStatus = Save(content, currentUserId);
+        ContentEditingOperationStatus operationStatus = Save(media, currentUserId);
         return operationStatus == ContentEditingOperationStatus.Success
-            ? Attempt.SucceedWithStatus(ContentEditingOperationStatus.Success, content)
-            : Attempt.FailWithStatus(operationStatus, content);
+            ? Attempt.SucceedWithStatus(ContentEditingOperationStatus.Success, media)
+            : Attempt.FailWithStatus(operationStatus, media);
     }
 
-    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveToRecycleBinAsync(Guid id, Guid userKey)
-    {
-        var currentUserId = await GetUserIdAsync(userKey);
-        return await HandleDeletionAsync(id, media => ContentService.MoveToRecycleBin(media, currentUserId).Result, false);
-    }
+    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveToRecycleBinAsync(Guid key, Guid userKey)
+        => await HandleMoveToRecycleBinAsync(key, userKey);
 
-    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> DeleteAsync(Guid id, Guid userKey)
-    {
-        var currentUserId = await GetUserIdAsync(userKey);
-        return await HandleDeletionAsync(id, media => ContentService.Delete(media, currentUserId).Result, true);
-    }
+    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> DeleteAsync(Guid key, Guid userKey)
+        => await HandleDeleteAsync(key, userKey);
+
+    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveAsync(Guid key, Guid? parentKey, Guid userKey)
+        => await HandleMoveAsync(key, parentKey, userKey);
 
-    public async Task<Attempt<IMedia?, ContentEditingOperationStatus>> MoveAsync(Guid id, Guid? parentId, Guid userKey)
+    public async Task<ContentEditingOperationStatus> SortAsync(Guid? parentKey, IEnumerable<SortingModel> sortingModels, Guid userKey)
+        => await HandleSortAsync(parentKey, sortingModels, userKey);
+
+    protected override IMedia New(string? name, int parentId, IMediaType mediaType)
+        => new Models.Media(name, parentId, mediaType);
+
+    protected override OperationResult? Move(IMedia media, int newParentId, int userId)
+        => ContentService.Move(media, newParentId, userId).Result;
+
+    protected override IMedia? Copy(IMedia media, int newParentId, bool relateToOriginal, bool includeDescendants, int userId)
+        => null; // copy is not supported for media
```

Comment:
``` suggestion
        => throw new NotImplementedException("Copy is not supported for media");
```
I like just failing fast here 🚀

Reply:
I'm all for it 💥

Reply:
I would prefer NotSupportedException. 

NotImplementedException is typically highlighted like TODOs in IDEs

Reply:
I can't amend it at this point, I'm sat with a trillion changes in another branch. @Zeegaan can you update the code before merging?

Reply:
I guess github can help here :), I'll take a look

Reply:
Oh, need to update OpenAPI.json anyway